Guide/src/dev_guide/dev_tools/xflowey.md

```diff
@@ -10,13 +10,753 @@ for writing maintainable, cross-platform automation.
+
+---
+
+## Node Design Philosophy
+
+Flowey nodes are designed around several key principles:
+
+### 1. Composability
+
+Nodes should be reusable building blocks that can be combined to build complex
+workflows. Each node should have a single, well-defined responsibility.
+
+❌ **Bad**: A node that "builds and tests the project"  
+✅ **Good**: Separate nodes for "build project" and "run tests"
+
+### 2. Explicit Dependencies
+
+Dependencies between steps should be explicit through variables, not implicit
+through side effects.
+
+❌ **Bad**: Assuming a tool is already installed  
+✅ **Good**: Taking a `ReadVar<SideEffect>` that proves installation happened
+
+### 3. Backend Abstraction
+
+Nodes should work across all backends when possible. Backend-specific behavior
+should be isolated and documented.
+
+### 4. Separation of Concerns
+
+Keep node definition (request types, dependencies) separate from step
+implementation (runtime logic):
+
+- **Node definition**: What the node does, what it depends on
+- **Step implementation**: How it does it
+
+---
+
+## Common Patterns
+
+### Request Aggregation and Validation
+
+When a FlowNode receives multiple requests, it often needs to ensure certain values are consistent across all requests while collecting others. The `same_across_all_reqs` helper function simplifies this pattern by validating that a value is identical across all requests.
+
+**Key concepts:**
+- Iterate through all requests and separate them by type
+- Use `same_across_all_reqs` to validate values that must be consistent
+- Collect values that can have multiple instances (like output variables)
+- Validate that required values were provided
+
+For a complete example, see the [`same_across_all_reqs` documentation](https://openvmm.dev/rustdoc/linux/flowey_core/node/user_facing/fn.same_across_all_reqs.html).
+
+### Conditional Execution Based on Backend/Platform
+
+Nodes can query the current backend and platform to emit platform-specific or backend-specific steps. This allows nodes to adapt their behavior based on the execution environment.
+
+**Key concepts:**
+- Use `ctx.backend()` to check if running locally, on ADO, or on GitHub Actions
+- Use `ctx.platform()` to check the operating system (Windows, Linux, macOS)
+- Use `ctx.arch()` to check the architecture (x86_64, Aarch64)
+- Emit different steps or use different tool configurations based on these values
+
+**When to use:**
+- Installing platform-specific tools or dependencies
+- Using different commands on Windows vs Unix systems
+- Optimizing for local development vs CI environments
+
+For more on backend and platform APIs, see the [`NodeCtx` documentation](https://openvmm.dev/rustdoc/linux/flowey_core/node/struct.NodeCtx.html).
+
+## Artifacts
+
+Artifacts enable typed data transfer between jobs with automatic dependency management, abstracting away CI system complexities like name collisions and manual job ordering.
+
+### Typed vs Untyped Artifacts
```

Comment:
this section needs to be proof-read by a human. there's some hallucinated info here (e.g: new_artifact returns a tuple of handles, and its those handles + the job-level API that convert them into read/write vars which are the things that link artifacts together)